add-kml-layer-with-network-links/src/main/java/com/esri/arcgismaps/sample/addkmllayerwithnetworklinks/components/SceneViewModel.kt

```diff
@@ -0,0 +1,74 @@
+/* Copyright 2024 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.addkmllayerwithnetworklinks.components
+
+import kotlinx.coroutines.launch
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+
+import com.arcgismaps.mapping.ArcGISScene
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.kml.KmlDataset
+import com.arcgismaps.mapping.layers.KmlLayer
+import com.esri.arcgismaps.sample.addkmllayerwithnetworklinks.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+
+class SceneViewModel(application: Application) : AndroidViewModel(application) {
+
+    // create a KML dataset from a URL, then use it to create a KML layer
+    val kmlDataset = KmlDataset(application.getString(R.string.kml_dataset_url))
```

Comment:
Personally, I've come round to putting URLs just as hardcoded strings.

The string assets are theoretically for translation--so in my opinion this just makes the sample harder to read.